src/wrapper/jnienv.rs

```diff
@@ -1497,6 +1497,88 @@ impl<'a> JNIEnv<'a> {
         res
     }
 
+    /// Get a static field without checking the provided type against the actual
+    /// field.
+    #[allow(unused_unsafe)]
+    pub unsafe fn get_static_field_unsafe<T, U>(
+        &self,
+        class: T,
+        field: U,
+        ty: JavaType,
+    ) -> Result<JValue>
+    where
+        T: Desc<'a, JClass<'a>>,
+        U: Desc<'a, JFieldID<'a>>,
+    {
+        let class = class.lookup(self)?.into_inner();
+
+        let field_id = field.lookup(self)?.into_inner();
+
+        // TODO clean this up
+        Ok(match ty {
+            JavaType::Object(_) | JavaType::Array(_) => {
+                let obj: JObject = jni_call!(self.internal, GetStaticObjectField, class, field_id);
+                obj.into()
+            }
+            // JavaType::Object
+            JavaType::Method(_) => unimplemented!(),
```

Comment:
I agree but I'm not an expert, I copied it from `get_field_unsafe`. What about replacing it with `return Err(...)`?

Reply:
As for me, panic (or even `unreachable!`) is OK here, especially because this function is marked as `unsafe`.

Can such `Err` be handled meaningfully?..

Reply:
I think `panic` would be better for that's not a kind of error you ever need to handle: if it occurs, you definitely use an API incorrectly (or there is a serious bug in a library itself).